dts/bindings/serial/uart-rs485-controller.yaml

```diff
@@ -0,0 +1,36 @@
+# Common fields for uart-rs485 controllers
+
+include: uart-controller.yaml
+
+properties:
+  rs485-enabled:
+    type: boolean
+    description: Enable rs485 mode
+  rs485-de-gpios:
+    type: phandle-array
+    description: Output for contactor and relay 1
```

Comment:
I'm raising this from STM32 perspective, but I'd be surprised we're the only vendor supporting RS485 HW Support:
It should be easy for user to configure and select use of built-in, RS485 HW support, vs RS485 SW support.
`CONFIG_SERIAL_SUPPORT_RS485` isn't the correct way to make this configuration as:
1-Even when disabled HW RS485 is available and supported
2- you're selecting it by default (while IMO the built-in HW support should be the default)

I'd advise exploring the possibility to perform the HW/SW RS485 support selection through the use of `rs485-de-gpios` property. Being available, SW is used, otherwise user can rely on HW support